src/main/xerus/monstercat/tabs/TabCatalog.kt

```diff
@@ -54,12 +61,67 @@ class TabCatalog: TableTab() {
 		table.visibleLeafColumns.addListener(ListChangeListener {
 			it.next(); Settings.VISIBLECATALOGCOLUMNS.putMulti(*it.addedSubList.map { it.text }.toTypedArray())
 		})
+		
+		fun playTracks(add: Boolean) {
+			val selected = table.selectionModel.selectedItems ?: return
+			GlobalScope.launch {
+				if(!add)
+					Player.playTracks(getSongs(selected))
+				else
+					Playlist.addAll(getSongs(selected))
+			}
+		}
 		table.setOnMouseClicked { me ->
-			if(me.clickCount == 2) {
-				val selected = table.selectionModel.selectedItem ?: return@setOnMouseClicked
-				Player.play(selected[cols.findUnsafe("Track")].trim(), selected[cols.findUnsafe("Artist")])
+			when {
+				me.clickCount == 2 && me.button == MouseButton.PRIMARY -> playTracks(false)
+				me.button == MouseButton.MIDDLE -> playTracks(true)
+			}
+		}
+		table.setOnKeyPressed { ke ->
+			when(ke.code) {
+				KeyCode.ENTER -> playTracks(false)
+				KeyCode.PLUS, KeyCode.ADD -> playTracks(true)
+				else -> return@setOnKeyPressed
+			}
+		}
+		
+		table.selectionModel.selectionMode = SelectionMode.MULTIPLE
+		
+		val rightClickMenu = ContextMenu()
+		rightClickMenu.items.addAll(
+			MenuItem("Play") {
+				val selected = table.selectionModel.selectedItems
+				GlobalScope.launch {
+					Player.playTracks(getSongs(selected))
+				}
+			},
+			MenuItem("Add to playlist") {
+				val selected = table.selectionModel.selectedItems
+				GlobalScope.launch {
+					Playlist.addAll(getSongs(selected))
+				}
+			},
+			MenuItem("Play next") {
+				val selected = table.selectionModel.selectedItems
+				GlobalScope.launch {
+					Playlist.addAll(getSongs(selected), true)
+				}
+			},
+			SeparatorMenuItem(),
+			MenuItem("Select All") {
+				table.selectionModel.selectAll()
 			}
+		)
+		table.contextMenu = rightClickMenu
+	}
+	
+	private suspend fun getSongs(songList: ObservableList<List<String>>): ArrayList<Track> {
+		val tracklist = arrayListOf<Track>()
+		songList.forEach { item ->
+			APIUtils.find(item[cols.findUnsafe("Track")].trim(), item[cols.findUnsafe("Artist")])?.let { tracklist.add(it) }
```

Comment:
You should add a line break before the `?.let`, the dev shouldn't have to scroll to the end of the line to see the effect of the function.

Reply:
if(it >= currentIndex.value!!) it + 1 else it }

Reply:
8625a5e10fd986e85f1eddf493bfb373758d8c5c